packages/amazonq/src/app/inline/cursorUpdateManager.ts

```diff
@@ -0,0 +1,190 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { LanguageClient } from 'vscode-languageclient'
+import { getLogger } from 'aws-core-vscode/shared'
+import { globals } from 'aws-core-vscode/shared'
+import { AmazonQInlineCompletionItemProvider } from './completion'
+
+// Configuration section for cursor updates
+export const cursorUpdateConfigurationSection = 'aws.q.cursorUpdate'
+
+/**
+ * Interface for recording completion requests
+ */
+export interface ICursorUpdateRecorder {
+    recordCompletionRequest(): void
+}
+
+/**
+ * Manages periodic cursor position updates for Next Edit Prediction
+ */
+export class CursorUpdateManager implements vscode.Disposable, ICursorUpdateRecorder {
```

Comment:
this needs to be disposed when deactivate. needs to be stopped when auto trigger is not enabled

Reply:
remove the usage of this for now https://github.com/aws/aws-toolkit-vscode/pull/7576